paddle/fluid/framework/details/all_reduce_op_handle.cc

```diff
@@ -90,22 +100,25 @@ void NCCLAllReduceOpHandle::RunImpl() {
           call();
         }
       });
+#else
+      PADDLE_THROW("Not compiled with CUDA");
+#endif
     } else {  // Special handle CPU only Operator's gradient. Like CRF
       auto &trg = *this->local_scopes_[0]
                        ->FindVar(kLocalExecScopeName)
                        ->Get<Scope *>()
-                       ->Var()
+                       ->FindVar(out_var_handles[0]->name_)
                        ->GetMutable<framework::LoDTensor>();
 
       // Reduce All Tensor to trg in CPU
       ReduceLoDTensor func(lod_tensors, &trg);
       VisitDataType(ToDataType(lod_tensors[0]->type()), func);
 
-      for (size_t i = 0; i < local_scopes_.size(); ++i) {
+      for (size_t i = 1; i < local_scopes_.size(); ++i) {
```

Comment:
This is not correct on GPU-CPU mixture code.

Reply:
Yes, you are right. 
But in the current implementation of `InsertNCCLAllReduceOp`, the inputs and outputs' `Variable` are the same respectively.
https://github.com/PaddlePaddle/Paddle/blob/1cfd3cb13b0ea4bf091757d59ec24e021563518f/paddle/fluid/framework/details/multi_devices_graph_builder.cc#L330-L340

So only to say, it doesn't cause error currently.